### DISCUSSION TITLE
Saving bookmarks in workspace

### DISCUSSION THREAD

**Discussion by dr5274** [alefragnani/vscode-bookmarks#581] [Ideas]
I searched and didn't see this, so thought I'd ask....is there any way to save bookmarks in a workspace?? I see you have the ability to save them at the repo level, but our repo is worked on by nearly 100 devs, so that doesn't work for us. Also, as we work on feature/fix branches, we typically save a workspace per branch, which allows us to open files relative to the work. Saving bookmarks relative to the work would also be very beneficial.

THANKS

**Comment by alefragnani** [comment]
Hi @dr5274 ,

No, this is not possible. 

Bookmarks are stored relative to the project (folder/repo), whether _locally_ (`.vscode/bookmarks.json` file) or in VSCode internals. Workspaces, on the other hand, may not be located within its containing folders, so the bookmarks would end up disconnected to the workspace itself.

There is however, an open request to global Bookmarks #287 , which means the bookmarks should end up being _per file_ instead of _per folder_. Maybe, this could fit your needs.

Hope this helps

**Comment by ez4gis** [comment]
First of all, thanks for providing such a useful extension! 

>...
> is there any way to save bookmarks in a workspace?
>...
> as we work on feature/fix branches, we typically save a workspace per branch, which allows us to open files relative to the work. 

+1  👍 

I would love to see this getting supported in the future too, for exactly the same reasons.  

I noticed that some extension such as "vscode-favorites" allowing storing settings/info inside the workspace file itself ("<workspace-name>.code-workspace") in json format. If that's possible with "vscode-bookmarks", it would be super helpful!

**Reply by alefragnani** [reply]
Hi @ez4gis ,

Thank you for your kind words. I'm glad to know you like the extension.

I'm starting to plan the next releases of the extension. Let's see where this could fit.
